contracts/liquidityMining/KyberRewardLocker.sol

```diff
@@ -221,38 +211,43 @@ contract KyberRewardLocker is IKyberRewardLocker, PermissionAdmin {
   {
     VestingSchedules storage schedules = accountVestingSchedules[msg.sender][token];
     uint256 totalVesting = 0;
-    uint256 totalSlashing = 0;
     for (uint256 i = 0; i < indexes.length; i++) {
       VestingSchedule memory schedule = schedules.data[indexes[i]];
       if (schedule.quantity == 0) {
         continue;
       }
       uint256 vestQuantity = _getVestingQuantity(
         schedule.quantity,
-        schedule.startTime,
-        schedule.endTime
+        schedule.startBlock,
+        schedule.endBlock
       );
       if (vestQuantity == 0) {
         continue;
       }
       totalVesting = totalVesting.add(vestQuantity);
-      totalSlashing = totalSlashing.add(schedule.quantity - vestQuantity);
-      // clear data after vesting
-      schedules.data[i].quantity = 0;
+
+      if (vestQuantity == uint256(schedule.quantity)) {
+        schedules.data[i].quantity = 0;
+      } else {
+        schedules.data[i] = VestingSchedule({
+          startBlock: _blockNumber().toUint64(),
```

Comment:
should it be better if we have a fixed startBlock so we have record of when it is claimed from FairLaunch?  having a `vestedQuantity` value and only need to update it whenever user vested

Reply:
This way is more efficiency for storage cost and computing, right?

Reply:
About UX, for example, the startBlock is 100, the endBlock is 150 and amount is 100 knc
user has vested 50 knc and the current block is 130.
It will be hard for front-end developers, right?